src/main/java/org/folio/service/mapping/profiles/MappingProfileService.java

```diff
@@ -0,0 +1,97 @@
+import static org.folio.service.mapping.profiles.RecordType.ITEM;
+import static org.folio.service.mapping.profiles.TransformationConfig.ELECTRONIC_ACCESS_LINK_TEXT;
+import static org.folio.service.mapping.profiles.TransformationConfig.ELECTRONIC_ACCESS_MATERIALS_SPECIFICATION;
+import static org.folio.service.mapping.profiles.TransformationConfig.ELECTRONIC_ACCESS_PUBLIC_NOTE;
+import static org.folio.service.mapping.profiles.TransformationConfig.ELECTRONIC_ACCESS_RELATIONSHIP_ID;
+import static org.folio.service.mapping.profiles.TransformationConfig.ELECTRONIC_ACCESS_URI;
+import static org.folio.service.mapping.profiles.TransformationConfig.MATERIAL_TYPE_ID;
+import static org.folio.service.mapping.profiles.TransformationConfig.PERMANENT_LOCATION_ID;
+
+public class MappingProfileService {
+
+  private static final Set<TransformationConfig> itemTransformationConfigs = EnumSet.of(
+    MATERIAL_TYPE_ID,
+    ELECTRONIC_ACCESS_LINK_TEXT,
+    ELECTRONIC_ACCESS_MATERIALS_SPECIFICATION,
+    ELECTRONIC_ACCESS_PUBLIC_NOTE,
+    ELECTRONIC_ACCESS_URI,
+    ELECTRONIC_ACCESS_RELATIONSHIP_ID
+  );
+
+  private static final Set<TransformationConfig> holdingsTransformationConfigs = EnumSet.of(
+    PERMANENT_LOCATION_ID,
+    ELECTRONIC_ACCESS_LINK_TEXT,
+    ELECTRONIC_ACCESS_MATERIALS_SPECIFICATION,
+    ELECTRONIC_ACCESS_PUBLIC_NOTE,
+    ELECTRONIC_ACCESS_URI,
+    ELECTRONIC_ACCESS_RELATIONSHIP_ID
+  );
+
+  public static final String SETTINGS_NAME_KEY = "name";
+
+  @Autowired
+  private PathBuilder pathBuilder;
+  @Autowired
+  private TrasnformationNameFormatter nameFormatter;
+
+  public List<MappingProfileTransformation> getMappingProfileTrasnformations(OkapiConnectionParams okapiConnectionParams) throws IOException {
+    List<MappingProfileTransformation> fieldNames = new ArrayList<>();
+    fieldNames.addAll(buildTransformations(ITEM, itemTransformationConfigs, okapiConnectionParams));
+    fieldNames.addAll(buildTransformations(HOLDINGS, holdingsTransformationConfigs, okapiConnectionParams));
+    return fieldNames;
+  }
+
+  private List<MappingProfileTransformation> buildTransformations(RecordType recordType, Set<TransformationConfig> transformationConfigs, OkapiConnectionParams okapiConnectionParams) {
+    List<MappingProfileTransformation> fieldNames = new ArrayList<>();
+    for (TransformationConfig transformationConfig : transformationConfigs) {
+      if (transformationConfig.isSettings()) {
+        List<MappingProfileTransformation> settingsFieldNames = buildTransformationsBySettings(recordType, okapiConnectionParams, transformationConfig);
+        fieldNames.addAll(settingsFieldNames);
+      } else {
+        fieldNames.add(buildSimpleTransformation(recordType, transformationConfig));
+      }
+    }
+    return fieldNames;
+  }
+
+  private List<MappingProfileTransformation> buildTransformationsBySettings(RecordType recordType, OkapiConnectionParams okapiConnectionParams, TransformationConfig transformationConfig) {
```

Comment:
I confirmed with Magda today, URL Relationship is not needed, so we can skip all the implementation  for Q2. WE can keep this PR though for further reference. We can just simply return the fixed values